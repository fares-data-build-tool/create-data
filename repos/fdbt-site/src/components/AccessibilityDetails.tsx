import React, { ReactElement } from 'react';

interface ContactProps {
    supportEmail: string;
}

const AccessibilityDetails = ({ supportEmail }: ContactProps): ReactElement => (
    <details className="govuk-details margin-top">
        <summary className="govuk-details__summary">
            <span className="govuk-details__summary-text">Request an accessible format.</span>
        </summary>
        <div className="govuk-details__text">
            If you use assistive technology (such as a screen reader) and need a version of this document in a more
            accessible format, please email <a href={`mailto:${supportEmail}`}>{supportEmail}</a>. Please tell us what
            format you need. It will help us if you say what assistive technology you use.
        </div>
    </details>
);

export default AccessibilityDetails;
