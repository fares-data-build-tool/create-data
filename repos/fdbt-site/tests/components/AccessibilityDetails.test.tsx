import { shallow } from 'enzyme';
import React from 'react';
import AccessibilityDetails from '../../src/components/AccessibilityDetails';

describe('AccessibilityDetails', () => {
    it('should render the AccessibilityDetails component', () => {
        const wrapper = shallow(<AccessibilityDetails supportEmail="test@gmail.com" />);
        expect(wrapper).toMatchSnapshot();
    });
});
