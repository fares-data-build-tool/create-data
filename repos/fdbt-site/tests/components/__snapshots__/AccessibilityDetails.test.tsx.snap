// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessibilityDetails should render the AccessibilityDetails component 1`] = `
<details
  className="govuk-details margin-top"
>
  <summary
    className="govuk-details__summary"
  >
    <span
      className="govuk-details__summary-text"
    >
      Request an accessible format.
    </span>
  </summary>
  <div
    className="govuk-details__text"
  >
    If you use assistive technology (such as a screen reader) and need a version of this document in a more accessible format, please email 
    <a
      href="mailto:test@gmail.com"
    >
      test@gmail.com
    </a>
    . Please tell us what format you need. It will help us if you say what assistive technology you use.
  </div>
</details>
`;
